### ISSUE TITLE
Command Line Parameters overriding is not working on windows machine

### ISSUE THREAD

**Issue by sarangan12** [Azure/autorest#4236]
I executed the following command in the mac machine:

```
autorest --typescript --allow-insecure-connection=true --is-test-package=true --license-header=true ./.tmp/specs/specification/sql/resource-manager/readme.md --clear-output-folder=true --output-folder=/Users/saranganrajamanickam/Projects/autorest.typescript/test/smoke/generated/sql-resource-manager --use=. --package-name=sql-resource-manager --modelerfour.lenient-model-deduplication
```

The above command executes successfully. Now I am executing the equivalent command in windows machine as:

```
autorest --typescript --allow-insecure-connection=true --is-test-package=true --license-header=true --clear-output-folder=true --use=. --package-name=sql-resource-manager --modelerfour.lenient-model-deduplication ./.tmp/specs/specification/sql/resource-manager/readme.md --output-folder=.\test\smoke\generated\sql-resource-manager
```

or

```
autorest --typescript --allow-insecure-connection=true --is-test-package=true --license-header=true --clear-output-folder=true  --use=C:\Users\sarajama\Projects\autorest.typescript --package-name=sql-resource-manager --modelerfour.lenient-model-deduplication C:\Users\sarajama\Projects\autorest.typescript\.tmp\specs\specification\sql\resource-manager\readme.md --output-folder=C:\Users\sarajama\Projects\temp
```

The above commands finish successfully without any errors. The ouput is:

```
............
............
WARNING (PreCheck/PropertyRedeclarationWarning): Schema 'FirewallRule' has a property 'type' that is already declared the parent schema 'Resource' but isn't significantly different. The property has been removed from FirewallRule

WARNING (PreCheck/PropertyRedeclarationWarning): Schema 'ReplicationLink' has a property 'type' that is already declared the parent schema 'Resource' but isn't significantly different. The property has been removed from ReplicationLink

WARNING (PreNamer/DeduplicateName): Deduplicating schema name: 'SecurityAlertPolicyName' -> 'SecurityAlertPolicyNameAutoGenerated'

WARNING (PreNamer/DeduplicateName): Deduplicating schema name: 'PrivateLinkServiceConnectionStateProperty' -> 'PrivateLinkServiceConnectionStatePropertyAutoGenerated'
INFORMATION: Autorest.Typescript took 94616ms
[192.32 s] Generation Complete
```

But, the output is is not generated in the value specified in the `--output-folder`. On further analysis, I found that the SDK is generated in the location of the readme.md file itself with the path: `C:\Users\sarajama\Projects\autorest.typescript\.tmp\specs\specification\sql\resource-manager\$(typescript-sdks-folder)\sdk\sql\arm-sql`. 

This error does not happen in the mac machine. On further analysis, I was able to find that the property `output-folder` is specified both in the configuration file and command line. The expectation is that the value provided in the command line will be used. (That is happening in Mac machine). But, it is not happening on Windows machine. (It used to work earlier) Please fix it at the earliest. 

@timotheeguerin FYI

**Comment by timotheeguerin** [comment]
Believe this is because it is under the typescript entry
```

``` yaml $(typescript)
typescript:
  azure-arm: true
  package-name: "@azure/arm-sql"
  output-folder: "$(typescript-sdks-folder)/sdk/sql/arm-sql"
  clear-output-folder: true
  generate-metadata: true
```

```

There is nothing that can be done right now. This is by design those group override the child property per plugin(typescript plugin).

**Comment by sarangan12** [comment]
@timotheeguerin Thanks for the clarification. I coul remove the `typescript:` from the configuration file. But, my question is why is this causing issue only in WIndows machine? This works fine for the Mac/Linux machine.

**Comment by sarangan12** [comment]
@timotheeguerin Any update on this?

**Comment by sarangan12** [comment]
@timotheeguerin Any update on this?

**Comment by timotheeguerin** [comment]
Can you check which version of core you are using on both system. I can't repro this issue on the latest version of core. `output-folder` is a special case to what I described above.

**Comment by sarangan12** [comment]
My output shows:

```
autorest.cmd --typescript --allow-insecure-connection=true --is-test-package=true --license-header=true ./.tmp/specs/specification/msi/resource-manager/readme.md --clear-output-folder=true --output-folder=C:\Users\sarajama\Projects\autorest.typescript\test\smoke\generated\msi-resource-manager --use=. --package-name=msi-resource-manager
AutoRest code generation utility [cli version: 3.1.1; node: v14.15.4, max-memory: 4096 MB]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest

There is a new version of AutoRest available (3.3.2).
 > You can install the newer version with with npm install -g autorest@latest

   Loading AutoRest core      'C:\Users\sarajama\.autorest\@autorest_core@3.5.1\node_modules\@autorest\core\dist' (3.5.1)
INFORMATION: > Loading local AutoRest extension '@autorest/typescript' (C:\Users\sarajama\Projects\autorest.typescript)
INFORMATION: > Loading AutoRest extension '@autorest/modelerfour' (4.19.3->4.19.3)
```

I am still able to get this on my windows machine.

**Comment by timotheeguerin** [comment]
Can you share the full log with `--debug --verbose --output-artifact:configuration.yaml` and share the configuration.yaml file created in the output dir.

Could you see also if you can create a simple repro with a basic readme+spec
